apps/backup_and_restore/backup_and_restore.py

```diff
@@ -0,0 +1,431 @@
+                    # match index_id on the target
+                    batch.add_reference(
+                      from_object_uuid=str(uuid.UUID(int=i)),
+                      from_object_class_name=class_name,
+                      from_property_name=f'to_{other}',
+                      to_object_uuid=str(uuid.UUID(int=i)),
+                      to_object_class_name=other,
+                    )
+
+def delete_records(client: weaviate.Client, class_name):
+    client.batch.delete_objects(
+        class_name=class_name,
+        where={
+            'operator': 'Equal',
+            'path': ['should_be_deleted'],
+            'valueBoolean': True
+        },
+        output='minimal',
+        dry_run=False,
+    )
+
+def fatal(msg):
+    logger.error(msg)
+    sys.exit(1)
+
+def success(msg):
+    logger.success(msg)
+
+def validate_dataset(client: weaviate.Client, class_name, expected_count=0):
+    # the filter removes 1/4 of elements
+    expected_filtered_count = expected_count * 3 / 4
+
+    result = client.query.aggregate(class_name) \
+        .with_fields('meta { count }') \
+        .with_fields("index_id {count}") \
+        .do()
+
+    logger.info("Aggregation without filters:")
+    total_count = result['data']['Aggregate'][class_name][0]['meta']['count']
+    prop_count = result['data']['Aggregate'][class_name][0]['index_id']['count']
+
+    if total_count != expected_count:
+        fatal(f"{class_name}: got {total_count} objects, wanted {expected_count}")
+    else:
+        success(f"{class_name}: got {total_count} objects, wanted {expected_count}")
+
+    if prop_count != expected_count:
+        fatal(f"{class_name}: got {prop_count} props, wanted {expected_count}")
+    else:
+        success(f"{class_name}: got {prop_count} props, wanted {expected_count}")
+
+    logger.info("Aggregation with filters")
+    result = client.query.aggregate(class_name) \
+        .with_where({'operator': 'Equal', 'valueBoolean':False, 'path':["is_divisible_by_four"]}) \
+        .with_fields('meta { count }') \
+        .with_fields("index_id {count}") \
+        .do()
+
+    total_count = result['data']['Aggregate'][class_name][0]['meta']['count']
+    prop_count = result['data']['Aggregate'][class_name][0]['index_id']['count']
+
+    if total_count != expected_filtered_count:
+        fatal(f"{class_name}: got {total_count} objects, wanted {expected_filtered_count}")
+    else:
+        success(f"{class_name}: got {total_count} objects, wanted {expected_filtered_count}")
+
+    if prop_count != expected_filtered_count:
+        fatal(f"{class_name}: got {prop_count} props, wanted {expected_filtered_count}")
+    else:
+        success(f"{class_name}: got {prop_count} props, wanted {expected_filtered_count}")
+
+def sample_range(start, end, size):
+    return enumerate(random.sample(range(start, end), size))
+
+def validate_stage(client: weaviate.Client, class_name, start=0, end=100_000, stage="stage_0", all_classes=[]):
```

Comment:
here is a second case of [] as default parameter

Reply:
Oh good catch. I'll fix it and push with `[ci skip]` as not to spam Travis with all the builds again.